include/ignition/gui/qml/PlottingInterface.qml

```diff
@@ -0,0 +1,556 @@
+  anchors.fill: parent
+  color: (Material.theme == Material.Light) ? Material.color(Material.Grey,Material.Shade100) : "transparent"
+
+  property var charts: ({})
+  property int mainChartID: 1
+
+  property int idIncrementor: 0
+
+  property bool multiChartsMode: false
+
+  // Horizonal Layout to hold multi charts (small charts)
+  Rectangle {
+      id: rowCharts
+      width: parent.width
+      height: (multiChartsMode) ? 150 : 0
+      color: (Material.theme == Material.Light) ? Material.color(Material.Grey,Material.Shade200)
+                                                : Material.color(Material.BlueGrey, Material.Shade800)
+      // Make it Scrolable
+      ScrollView {
+          anchors.fill: parent
+          ScrollBar.horizontal.policy: ScrollBar.AsNeeded
+          ScrollBar.vertical.policy: ScrollBar.AlwaysOff
+          clip: true
+          // Horizontal Layout for the Charts
+          Row {
+              id:rowChartsLayout
+              anchors.fill: parent
+              spacing: 10
+          }
+      }
+  }
+
+  // Vertical Layout to hold Main Charts (1 or 2 charts)
+  Column {
+      id : chartsLayout
+
+      width: parent.width
+      anchors.topMargin: 10
+      anchors.top: rowCharts.bottom
+      anchors.bottom: parent.bottom
+      property int heightFactor: 0
+  }
+
+
+  Rectangle {
+      id : addBtn
+
+      anchors.right: openExport.left
+      anchors.top: parent.top
+      anchors.margins: 15
+
+      width: 40
+      height: 40
+      radius: width/2
+      color: Material.accentColor
+      Text {
+          text:  "+"
+          font.weight: Font.Bold
+          font.pixelSize: parent.width/2
+          color: "white"
+          anchors.centerIn: parent
+      }
+
+      MouseArea {
+          id: mouseAddBtn
+          anchors.fill: parent
+          hoverEnabled: true
+          onEntered: { addBtn.opacity = 0.8; cursorShape = Qt.PointingHandCursor; }
+          onExited: { addBtn.opacity = 1; cursorShape =  Qt.ArrowCursor; }
+
+          onClicked: {
+              if (Object.keys(charts).length == 2)
+              {
+                  multiChartsMode = true;
+                  var firstChart = true;
+                  for (var i = 0; i < chartsLayout.children.length; i++)
+                  {
+                      // skip the first one
+                      if (firstChart)
+                      {
+                          firstChart = false;
+                          // make that chart has a full size
+                          chartsLayout.heightFactor = 1
+                          continue;
+                      }
+                      var chart = chartsLayout.children[i];
+                      reallocateChart(chart);
+                  }
+              } // end if
+
+              addChart();
+          }
+      }
+
+      ToolTip.text: "Add Chart"
+      ToolTip.delay: 500
+      ToolTip.timeout: 1000
+      ToolTip.visible: mouseAddBtn.containsMouse
+  }
+
+  function addChart()
```

Comment:
Also remember to document all functions. And I recommend moving all functions and properties to the top, before any UI items.